src/lib/gundb/UserStorageClass.js

```diff
@@ -726,7 +731,8 @@ export class UserStorage {
 
     if (!inviteCode) {
       const { data } = await API.getUserFromW3ByToken(_token).catch(e => {
-        logger.warn('failed fetching w3 user', { e })
+        const errMsg = getApiErrorText(e)
```

Comment:
see https://github.com/GoodDollar/GoodDAPP/pull/2330#discussion_r477410377

Reply:
The error.message will be changed inside the getErrorMessage method

Reply:
Check the getErrorMessage method

Reply:
or you worried about something else?

Reply:
'e' thrown from the axios in most cases isn't an exception. 

It's an exception in two cases only:
- when there's no response from the backend
- when there're network issues

In other cases interceptor throws the raw or JSON-parsed http response body
- if server returns string - it will be the string e.g. 'Request failed with status code 500'
- if server returns json - it will be an plain object. For requests performing to GoodServer this object will contain { error: <message string> }

So, we couldn't log 'e' itself, we have to create new Error() and we have to do it excactly in the place we're logging to keep the stack trace. We could skip the check for isError() ti simplify the code.

So please do

```javascript
someApiCall().catch(e => {
  const message = getErrorMessage(e)
  const exception =  new Error(message)


  log.error('api error happens', message, exception)
```

everywhere we calling API (except the face verification where i already processing the e object, let's keep it as is. in the login service you could replace my latest fix with getErrorMessage)